src/schemas/21-686C/schema.js

```diff
@@ -1,18 +1,482 @@
+      'enum': [countryUSA, countryMilitary, countryNotInList].concat(countriesForeign)
+    }
+  }
+}
+
+const postalCode = {
+  type: 'string',
+  maxLength: 10,
+  pattern: '^[0-9]{5,10}$'
+};
+
+const domesticAddress = {
+  required: [...commonAddressFields.required, 'state', 'postalCode'],
+  properties: {
+    ...commonAddressFields.properties,
+    addressType: {
+      type: 'string',
+      'enum': ['DOMESTIC'],
+      default: 'DOMESTIC'  
+    },
+    state: {
+      type: 'string',
+      maxLength: 50,
+      'enum': states.map(state => state.value),
+      enumNames: states.map(state => state.label)
+    },
+    postalCode,
+    country: {
+      type: 'string',
+      maxLength: 50,
+      default: countryUSA
+    }
+  },
+  additionalProperties: false
+}
+
+const militaryAddress = {
+  required: [...commonAddressFields.required, 'postOffice', 'postalType', 'postalCode'],
+  properties: {
+    ...commonAddressFields.properties, 
+    addressType: {
+      type: 'string',
+      enum: ['MILITARY'],
+      default: 'MILITARY'  
+    },
+    postOffice: {
+      type: 'string',
+      'enum': [
+        'APO',
+        'DPO',
+        'FPO'
+      ],
+      enumNames: [
+        'Ashore Post Office',
+        'Diplomatic Post Office',
+        'Fleet Post Office'
+      ]
+      
+    },
+    postalType: {
+      type: 'string',
+      'enum': [
+        'AA',
+        'AE',
+        'AP'
+      ],
+      enumNames: [
+        'Armed Forces Americas (except Canada)',
+        'Armed Forces Europe, Middle East, Canada & Africa',
+        'Armed Forces Pacific'
+      ]
+    },
+    postalCode
+  },
+  additionalProperties: false
+}
+
+
+const internationalAddressDropDown = {
+  required: [...commonAddressFields.required],
+  properties: {
+    ...commonAddressFields.properties,
+    addressType: {
+      type: 'string',
+      enum: ['INTERNATIONAL'],
+      default: 'INTERNATIONAL'
+    },
+    country: {
+      type: 'string',
+      'enum': countriesForeign
+    }
+  },
+  additionalProperties: false
+}
+
+const internationalAddressText = {
+  required: [...commonAddressFields.required, 'countryText'],
+  properties: {
+    ...commonAddressFields.properties,
+    addressType: {
+      type: 'string',
+      enum: ['INTERNATIONAL'],
+      default: 'INTERNATIONAL'
+    },
+    country: {
+      type: 'string',
+      'enum': [countryNotInList],
+      default: countryNotInList
+    },
+    countryText: {
+      type: 'string',
+      maxLength: 50,
+      minLength: 1,
+      pattern: textRegex
+    }
+  },
+  additionalProperties: false
+}
+
+
+
+
 let schema = {
   $schema: 'http://json-schema.org/draft-04/schema#',
   title: 'DECLARATION OF STATUS OF DEPENDENTS',
   type: 'object',
   additionalProperties: false,
-  definitions: {},
+  definitions,
   properties: {
+    claimantAddress: {
+      type: 'object',
+      oneOf: [domesticAddress, militaryAddress, internationalAddressDropDown, internationalAddressText]
+    },
```

Comment:
```
{
  type: 'object',
  oneOf: [domesticAddress, militaryAddress, internationalAddressDropDown, internationalAddressText]
}
```
can this be put in a variable since it's used in multiple places?